### ISSUE TITLE
Crashing on Android

### ISSUE THREAD

**Issue by jacquesdev** [Instabug/Instabug-Android#224]
I updated to version 8.6.1 of the plugin (from 8.5.1). I uploaded to beta on the Google Play Store, and getting crash reports related to instabug

### SDK Version
8.6.1

### React Native, iOS and Android Versions
```
  React Native Environment Info:
    System:
      OS: macOS 10.14.6
      CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
      Memory: 4.61 GB / 16.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.15.3 - ~/.nvm/versions/node/v10.15.3/bin/node
      Yarn: 1.15.2 - ~/.yarn/bin/yarn
      npm: 6.10.2 - ~/.nvm/versions/node/v10.15.3/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.4, macOS 10.14, tvOS 12.4, watchOS 5.3
      Android SDK:
        API Levels: 28, 29
        Build Tools: 28.0.3, 29.0.2
        System Images: android-28 | Google APIs Intel x86 Atom
    IDEs:
      Android Studio: 3.4 AI-183.6156.11.34.5692245
      Xcode: 10.3/10G8 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.3 => 16.8.3 
      react-native: 0.59.9 => 0.59.9 
    npmGlobalPackages:
      react-native-cli: 2.0.1
      react-native-git-upgrade: 0.2.7
```
### Device Model
 Android 8.0.0
 Galaxy S7

```
APIBuildChecker.java line 6
com.instabug.library.APIBuildChecker.check

Fatal Exception: java.lang.RuntimeException
An error occurred while executing doInBackground()
Fatal Exception: java.lang.RuntimeException: An error occurred while executing doInBackground()
       at android.os.AsyncTask$3.done + 353(AsyncTask.java:353)
       at java.util.concurrent.FutureTask.finishCompletion + 383(FutureTask.java:383)
       at java.util.concurrent.FutureTask.setException + 252(FutureTask.java:252)
       at java.util.concurrent.FutureTask.run + 271(FutureTask.java:271)
       at java.util.concurrent.ThreadPoolExecutor.runWorker + 1162(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run + 636(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run + 764(Thread.java:764)
```

**Comment by hugosvent** [comment]
this happened to me as well.
since I'm upgrading my instabug from 8.4.0 to 8.6.0

**Comment by jacquesdev** [comment]
Does anyone know how long a fix for this will take? It's holding up my play store release.

**Comment by amrbrkt** [comment]
Hey @jacquesdev, 
Would you please try to use the latest version (8.6.3) and let us know if you still facing any issues.

**Comment by CWftw** [comment]
@amrbrkt I am getting hundreds of the same crash reports on 3.6.3

**Comment by amrbrkt** [comment]
@CWftw do you mean 8.6.3?

**Comment by CWftw** [comment]
Yes, sorry

**Comment by amrbrkt** [comment]
Are trying to use one of instabug APIs in `AsyncTask`(off main thread), it might be executed before the initialization of instabug is done on the main thread. would you verify this?

**Comment by CWftw** [comment]
We setup Instabug in a Handler post in our application class, although this handler should be on the main thread. We only invoke via a click event and shake. We also call "setInvocationEvents" when the application starts in a Fragment

**Comment by amrbrkt** [comment]
@CWftw would you please, provide a sample project that simulates your implementation and produces the issue?

**Comment by CWftw** [comment]
@amrbrkt I didn't have time to do that. However, I have removed the initialization from the handler to be 100% sure it was initialized before any other calls were made and it seems to have made the above crashes go away. That being said - while I understand that this was not an intended use case - I don't believe that crashing user's apps is a good way to handle this. Instabug introduces new crashes in almost every release and strongly encourages developers to update causing more bugs than it could ever prevent, defeating the whole purpose of the platform.

**Comment by mzelzoghbi** [comment]
Hi @CWftw,
Please accept our apologies for the inconveniences we may have caused. I completely understand how that can be frustrating.
Please rest assured that we’re always working on improving the stability of our SDKs while introducing new helpful features. When Instabug is initialized in a way that’s different to how it should normally be initialized, sometimes unexpected behavior can occur. We will definitely be working on improving this as part of our efforts to improve the SDK stability.

Thank You

**Comment by mattetre** [comment]
We are experiencing a similar issue on Android platforms using `Instabug-React-Native` since we have updated to `v8.6.2`. We don't make any native calls to the instabug SDK we only access via Javascript.

We are looking to see if updating to `v8.6.3` would help us but are experiencing: https://github.com/Instabug/Instabug-React-Native/issues/387 during our upgrade

**Comment by mzelzoghbi** [comment]
@mattetre Can you please share with us the issues you are experiencing on v8.6.2?

**Comment by mattetre** [comment]
@mzelzoghbi We are seeing this reported on Google Play. It does not happen for everyone, just sometimes

![Screenshot 2019-09-11 at 12 09 54](https://user-images.githubusercontent.com/1724838/64692547-5438da00-d48d-11e9-998f-24457f514163.png)

**Comment by mzelzoghbi** [comment]
@mattetre These crashes are from `InstabugSDK-version`: **8.6.2**, right?

**Comment by mattetre** [comment]
Yes, react-native instabug `8.6.2`

**Comment by jacquesdev** [comment]
@mattetre - can you confirm if this works on 8.6.4?

@mzelzoghbi - can you confirm where my issue was fixed in 8.6.4, in in which commit. I am wary of updating to 8.6.4 and for this issue to start happening again. I have since realised my issue is in fact in this repo https://github.com/Instabug/Instabug-React-Native, but not sure if that just uses this one under the hood.

**Comment by mzelzoghbi** [comment]
@jacquesdev This issue has been fixed on Instabug native android SDK 8.7.0 and will be included on ReactNative release 8.7..0 planned to be within a week.
